test/unit-test/ota_utest.c

```diff
@@ -2229,6 +2281,33 @@ void test_OTA_RefreshWithSameJobDoc()
     refreshWithJobDoc( JOB_DOC_A, JOB_DOC_A );
 }
 
+void test_OTA_RefreshWithInvalidJobDoc()
+{
+    OtaEventMsg_t otaEvent = { 0 };
+
+    pOtaJobDoc = JOB_DOC_A;
+
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForFileBlock, OTA_GetState() );
+
+    /* Set the event send interface to a mock function that allows events to be sent continuously.
+     * We need this to go through the process of refreshing job doc. */
+    otaInterfaces.os.event.send = mockOSEventSend;
+
+    /* First send request job doc event while we're in progress, this should make OTA agent to
+     * to request job doc again and transit to waiting for job state. */
```

Comment:
I recommend updating this comment to something like this:
"Send the request job document event while the OTA Agent is already in the process of downloading a file. The OTA Agent should cancel the current job and begin waiting for the next job document."